pkg/fastforward/fastforward.go

```diff
@@ -80,18 +76,39 @@ func (f *FastForward) Run() error {
 		f.RepoSetDry(repo)
 	}
 
-	logrus.Infof("Checking if %q is a release branch", branch)
-	if isReleaseBranch := f.IsReleaseBranch(branch); !isReleaseBranch {
-		return errors.Errorf("%s is not a release branch", branch)
-	}
+	branch := f.options.Branch
+	if branch == "" {
+		logrus.Info("No release branch specified, finding the latest")
+		branch, err = f.RepoLatestReleaseBranch(repo)
+		if err != nil {
+			return errors.Wrap(err, "finding latest release branch")
+		}
 
-	logrus.Info("Checking if branch is available on the default remote")
-	branchExists, err := f.RepoHasRemoteBranch(repo, branch)
-	if err != nil {
-		return errors.Wrap(err, "checking if branch exists on the default remote")
-	}
-	if !branchExists {
-		return errors.New("branch does not exist on the default remote")
+		notRequired, err := f.noFastForwardRequired(repo, branch)
+		if err != nil {
+			return errors.Wrap(err, "check if fast forward is required")
+		}
+		if notRequired {
+			logrus.Infof(
+				"Fast forward not required because final tag already exists for latest release branch %s",
+				branch,
+			)
+			return nil
+		}
+	} else {
+		logrus.Infof("Checking if %q is a release branch", branch)
+		if isReleaseBranch := f.IsReleaseBranch(branch); !isReleaseBranch {
+			return errors.Errorf("%s is not a release branch", branch)
+		}
+
+		logrus.Info("Checking if branch is available on the default remote")
+		branchExists, err := f.RepoHasRemoteBranch(repo, branch)
+		if err != nil {
+			return errors.Wrap(err, "checking if branch exists on the default remote")
+		}
+		if !branchExists {
+			return errors.New("branch does not exist on the default remote")
+		}
```

Comment:
I'm curious here, why is there a different check logic when specifying the branch? I get it that we will be running it mostly in the latest branch and this completes the other case.
But are you thinking of a possible use for ff'ing other branches now?

Reply:
In theory the tool could be used to sync any branch combination. I’d wanted to preserve that while only adding automation for our additional use case.

I think we can use the manual path by ff-ing this cycle again, if we communicate it early enough (in the next week).

The automation can be used to verify that it works and could go live in 1.25. WDYT?